fake-koji/src/test/java/org/fakekoji/xmlrpc/server/core/FileReturningHandlerTest.java

```diff
@@ -2,26 +2,126 @@
+            "aarch64.shenandoah.jdk8u141.b16",
+            "aarch64.shenandoah.jdk8u141.b16.shenandoah.merge.2017.07.27.02",
+            "aarch64.shenandoah.jdk8u141.b16.shenandoah.merge.2017.07.26",
+            "aarch64.shenandoah.jdk8u131.b12",
+            "aarch64.shenandoah.jdk8u131.b12.shenandoah.merge.2017.04.20",
+            "aarch64.shenandoah.jdk8u131.b11",
+            "aarch64.shenandoah.jdk8u121.b14.shenandoah.merge.2017.03.23",
+            "aarch64.shenandoah.jdk8u121.b14.shenandoah.merge.2017.03.09",
+            "aarch64.shenandoah.jdk8u121.b14.shenandoah.merge.2017.03.08",
+            "aarch64.shenandoah.jdk8u121.b14.shenandoah.merge.2017.03.06",
+            "arch64.jdk8u131.b12",
+            "jdk8u999.b99.zgu.20160920",
+            "jdk8u162.b01",
+            "jdk8u162.b00",
+            "jdk8u152.b04",
+            "jdk8u152.b03",
+            "jdk8u152.b02",
+            "jdk8u152.b01",
+            "jdk8u144.b01",
+            "jdk8u141.b15",
+            "jdk8u122.b01",
+            "jdk8u122.b00",
+            "jdk8u121.b13",
+            "jdk8u.zgu.20160920aarch64.jdk8u144.b02"
+    };
+
+    private final String[] ojdk9files = new String[] {
+            "9.0.0.163",
+            "9.0.0.154",
+            "jdk.9.181",
+            "jdk.9.180",
+            "jdk.9.178",
+            "jdk.9.176",
+            "jdk.9.175",
+            "jdk.9.174",
+            "jdk.9.173",
+            "jdk.9.172",
+            "jdk.9.170",
+            "jdk.9.167",
+            "jdk.9.165",
+            "jdk.9.164",
+            "jdk.9.156"
+    };
+
+    private List<FileReturningHandler.FileInfo> getFileInfoList(String[] files) {
+        List<FileReturningHandler.FileInfo> list = new ArrayList<>();
+        long size = 1;
+        for (String str : files) {
+            list.add(new FileReturningHandler.FileInfo(str, null, null, size));
+            size++;
+        }
+        return list;
+    }
+
     @Test
-    public void testComparatorByVersion() throws Exception {
+    public void testComparatorByVersionOjdk7() throws Exception {
         Comparator<FileReturningHandler.FileInfo> c = new FileReturningHandler.ComparatorByVersion();
-        FileReturningHandler.FileInfo fi1 = new FileReturningHandler.FileInfo("name1", null, null, 1);
-        FileReturningHandler.FileInfo fi2 = new FileReturningHandler.FileInfo("name2", null, null, 2);
-        FileReturningHandler.FileInfo fi3 = new FileReturningHandler.FileInfo("name3", null, null, 3);
-        FileReturningHandler.FileInfo[] fi = new FileReturningHandler.FileInfo[]{
-            fi2, fi3, fi1
-        };
-        Arrays.sort(fi, c);
-        Assert.assertArrayEquals(
-                new FileReturningHandler.FileInfo[]{
-                    fi3, fi2, fi1
-                }, fi);
+        List<FileReturningHandler.FileInfo> list = getFileInfoList(ojdk7files);
+        List<FileReturningHandler.FileInfo> copy = new ArrayList<>(list);
+        Collections.shuffle(copy);
```

Comment:
Shuffle right?-)) Well +1 for imagination... It is not exactly a implementation of deterministic test.. but OK:)